sequence_processing_pipeline/NuQCJob.py

```diff
@@ -116,6 +118,63 @@ def __init__(self, fastq_root_dir, output_path, sample_sheet_path,
 
         self._validate_project_data()
 
+    def contains_bx_metadata(self):
+        # get list of raw compressed fastq files. only consider R1 and R2
+        # reads.
+
+        # Note that NuQCJob works across all projects in a sample-sheet. if
+        # there are more than one project in the sample-sheet and if one
+        # is a TellSeq project and one isn't then that would break an
+        # an assumption of this helper (one file is representative of all
+        # files.) However since tellseq requires a special sample-sheet, it
+        # can be assumed that all projects in a tellseq sample-sheet will be
+        # tellseq and therefore carry the TellSeq BX metadata. The inverse
+        # should also be true.
+
+        fastq_paths = glob(self.root_dir + '/*/*.fastq.gz')
```

Comment:
This seems like replicated logic from other changes above? Shouldn't there be a single way to find fastq files?

Reply:
Moot at this point.